aquarius/events/processors.py

```diff
@@ -345,17 +395,24 @@ def process(self):
             self._chain_id,
             txid,
             self.event.args.metaDataHash,
+            self._es_instance,
         )
         if not asset:
-            logger.info("Decrypter is not authorized. Failing gracefully.")
+            logger.info("Decrypt ddo failed.Failing gracefully.")
             return
 
         self.did = asset["id"]
         did, sender_address = self.did, self.sender_address
 
         permission = self.check_permission(sender_address)
         if not permission:
-            raise Exception("RBAC permission denied.")
+            error = "RBAC permission denied."
+            logger.info(error)
+            if self._es_instance:
+                self._es_instance.update_did_state(
```

Comment:
we keep doing this... I don't think there should be so much paste. Can we move it to a class function e.g. self.update_did_state(txid, error)?

then inside self._es_instance.update_did_state(self.event.address, self._chain_id, txid, False, error)?

at least it would make it more concise

Reply:
will solve it in another PR

Reply:
see https://github.com/oceanprotocol/aquarius/pull/959/commits/c17d8615e315c8f0cb8dc5b5b59836b16b2c3b20